### PR TITLE
LP-170 Add Logging Interceptor to log all requests to the api

### PR DESCRIPTION
**Pull request by markpit**
https://companieshouse.atlassian.net/browse/LP-170

Add Logging Interceptor to log all incoming requests to the api

### CODE REVIEW

File: src/test/java/uk/gov/companieshouse/limitedpartnershipsapi/utils/ApiLoggerTest.java [hunk 1]
```diff
@@ -41,7 +41,7 @@ void testInfoLoggingDoesNotModifyLogMap() {
 
     @Test
     void testInfoContextLoggingDoesNotModifyLogMap() {
-        ApiLogger.infoContext(CONTEXT, TEST_MESSAGE);
+        ApiLogger.infoContext(CONTEXT, TEST_MESSAGE, logMap);
```
Comment (mattch1) [inline src/test/java/uk/gov/companieshouse/limitedpartnershipsapi/utils/ApiLoggerTest.java:44]: Is it worth testing debug and error logging here as well?

Comment (markpit) [reply src/test/java/uk/gov/companieshouse/limitedpartnershipsapi/utils/ApiLoggerTest.java:44]: i haven't changed those, only added a new infoContext method

Comment (mattch1) [reply src/test/java/uk/gov/companieshouse/limitedpartnershipsapi/utils/ApiLoggerTest.java:44]: Ignore, we seem to have those.

**Review by mattch1 — APPROVED** [review]
LGTM

**Comment by ch-code-analysis** [comment]
CI: No security warnings found

**Comment by ch-code-analysis** [comment]
CI: Security warnings found!

File: src/main/java/uk/gov/companieshouse/limitedpartnershipsapi/config/InterceptorConfig.java [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+    public void addInterceptors(@NonNull InterceptorRegistry registry) {
+        addLoggingInterceptor(registry);
+    }
+
+    /**
+     * Interceptor that logs all calls to endpoints
+     * @param registry The spring interceptor registry
+     */
+    private void addLoggingInterceptor(InterceptorRegistry registry) {
+        registry.addInterceptor(loggingInterceptor);
```
Comment (mwestacott) [inline src/main/java/uk/gov/companieshouse/limitedpartnershipsapi/config/InterceptorConfig.java:34]: Could just move this line to 26 and not have the private method (with Javadoc). Save on LOC... Or do you think it makes code more readable?

Comment (markpit) [reply src/main/java/uk/gov/companieshouse/limitedpartnershipsapi/config/InterceptorConfig.java:34]: i can move it

**Comment by ch-code-analysis** [comment]
CI: No security warnings found

**Comment by ch-code-analysis** [comment]
CI: No security warnings found

**Review by mwestacott — APPROVED** [review]

**Review by HatiDev — APPROVED** [review]

**Pull request merged**